course_info/static/course_info/js/controllers/DetailsController.js

```diff
@@ -0,0 +1,110 @@
+// TODO - this is pretty much a copy of the people controller with some mods.
+// TODO - this still needs a lot of clean up, we don't need everything in here
+
+(function () {
+    var app = angular.module('CourseInfo');
+    app.controller('DetailsController', DetailsController);
+
+    function DetailsController($scope, $routeParams, courseInstances, $compile,
+                               djangoUrl, $http, $q, $log, $uibModal, $sce) {
+
+        dc = this;
+
+        dc.handleAjaxError = function (data, status, headers, config, statusText) {
+            $log.error('Error attempting to ' + config.method + ' ' + config.url +
+                ': ' + status + ' ' + statusText + ': ' + JSON.stringify(data));
+        };
+
+        dc.handleLookupResults = function(results) {
+            var courseResult = results[0];
+            var membersResult = results[1];
+
+            //check if the right data was obtained before storing it
+            if (courseResult.data.course_instance_id == dc.courseInstanceId) {
+                courseInstances.instances[courseResult.data.course_instance_id] = courseResult.data;
+                dc.courseInstance = dc.getFormattedCourseInstance(courseResult.data, membersResult.data)
+            } else {
+                $log.error(' CourseInstance record mismatch for id :'
+                    + dc.courseInstanceId + ',  fetched record for :' + courseResult.data.id);
+            }
+        };
+
+        dc.setCourseInstance = function (id) {
+
+            var course_url = djangoUrl.reverse(
+                'icommons_rest_api_proxy',
+                ['api/course/v2/course_instances/' + id + '/']);
+
+            var members_url = djangoUrl.reverse(
+                'icommons_rest_api_proxy',
+                ['api/course/v2/course_instances/'
+                + id + '/people/']);
+
+            var coursePromise = $http.get(course_url)
+                .error(dc.handleAjaxError);
+
+            var membersPromise = $http.get(members_url)
+                .error(dc.handleAjaxError);
```

Comment:
It would be great if the `coursePromise` could initiate display of the data table, and the `membersPromise` could fill in the people info separately, since the `people/` endpoint is slower (the people count section could show a loading indicator until it returned).